### ISSUE TITLE
Problems reading .env file with and without `python-dotenv`

### ISSUE THREAD

**Issue by lowercase00** [tiangolo/fastapi#3369]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Description

I have been experiencing quite a few issues using .env files. Coming from Flask, this used to be as simple as `env = os.getenv("APP_ENV")`, but in FastAPI I can't make it work. I initially thought the problem was with the way Poetry was handling virtual environments, but using the same virtual environment (created with Poetry) and creating a minimal Flask app, and a minimal FastAPI app, it seems that this has something to do with the way Flask handles this type of files (maybe?). 

I never used `python-dotenv` with Flask, but even with `python-dontenv` installed in my venv, and even with `from dotenv import find_dotenv` I can't get access to the env file.

On the example bellow, Flask's app prints the "dev", while in FastAPI I get "None".

### Example

```.env file (parent folder
APP_ENV="dev"
```

```Python
import os
from fastapi import FastAPI

app = FastAPI()

@app.get("/")
def read_root():
    env = os.getenv("APP_ENV")
    print(env)
    return {"Hello": "World"}
```

```Python
import os
from flask import Flask

app = Flask(__name__)

@app.route("/")
def read_root():
    env = os.getenv("APP_ENV")
    print(env)
    return {"Hello": "World"}
```

### Environment

* macOS 11.4
* FastAPI Version: 0.60
* Python 3.8 (pyenv)
* Poetry shell

**Comment by Mause** [comment]
Have you tried `load_dotenv`instead of `find_dotenv`?

**Comment by lowercase00** [comment]
@Mause indeed I was doing something wrong: `find_dotenv` returns the file path, and doesn't load the `.env` file. 
What works is `finding` the path, and loading the `load_dotenv`.

```python
env_file = find_dotenv()
load_dotenv(env_file)
```

Would be great to have this built-in.

**Comment by lsglucas** [comment]
I don't know if you had already used this approach or if you don't like it, but I suggest you set your .env variables using [Pydantic settings](https://pydantic-docs.helpmanual.io/usage/settings/#dotenv-env-support)

I like using Pydantic to set the correct `.env`, and other settings variables I might have, in my opinion, it's easier to set up and you'll have less repetitive code.

**Comment by boanergies** [comment]
I agree with the Pydantic settings recommendation, much much easier to handle this situation your in and also much easier to work with python-dotenv for development.

I'll show you how in my project I'm currently using it and maybe it will give some insight to you on how to do it simpler.

For my own use case I'm using [deta](www.deta.sh) which requires a project token when using the database. So I set up a pydantic settings to help validate that a project token has been set in the environment to be used when wanting to work with the deta database.

You'll see below that the `DatabaseSettings` class requires the deta project key which will automatically check that a 'DETA_PROJECT_KEY' variable has been set in the environment. The `Config` class inside of it is what enables python-dotenv to also be an option when declaring environment variables meaning you can drop a .env file with the environment variables listed without having to actually declare environment variables in your system. Which I don't about you, but I like the idea I don't need to mess with my system state when I'm developing code.

From there I merely instantiate the `DatabaseSettings` when creating the database and if everything is good then I'm able to create a `Deta` instance to get access to the database API.

Hopefully this helps!

```Python
from pydantic import BaseSettings, ValidationError


class DatabaseSettings(BaseSettings):
    deta_project_key: str

    class Config:
        env_file = ".env"


def get_database_settings():
    return DatabaseSettings()


def parse_database_validation_error(validation_error: ValidationError):
    error_string = ""
    for error in validation_error.errors():
        if error["type"] == "value_error.missing":
            for field in error["loc"]:
                error_string += f"ERROR: Environment Variable {field.upper()} was not set"
        else:
            raise Exception(
                f"ERROR: UNKNOWN ERROR VALIDATION OF SETTING FOUND! CREATE A PROPER ERROR MESSAGE FOR IT! {error}"
            )
    return error_string
```

```Python
DATABASE_NAMES = ["recipes", "inventory", "shopping-list-items"]


class DatabaseError(Exception):
    pass


class DatabaseSettingsError(Exception):
    pass


class DoesNotExistError(LookupError):
    pass


@dataclass
class Database:
    database_name: str
    database: Deta.Base = field(init=False)

    def __post_init__(self):
        try:
            settings = get_database_settings()
        except ValidationError as validation_error:
            raise DatabaseSettingsError(parse_database_validation_error(validation_error))
        if self.database_name not in DATABASE_NAMES:
            raise DatabaseError(
                f"ERROR: invalid database name given, choose from the following to connect to {DATABASE_NAMES}"
            )
        deta = Deta(settings.deta_project_key)
        self.database = deta.Base(self.database_name)
        self.util = self.database.util
```
